dessia_common/excel_reader.py

```diff
@@ -0,0 +1,503 @@
+        return instantiated_objects
+
+    def process_workbook(self):
+        """
+        Process the workbook's sheets and extract information.
+
+        This function iterates through each sheet in the workbook and gathers specific data from each sheet.
+        It collects module and class names along with attributes and associated data.
+
+        Returns:
+        dict: A dictionary containing information extracted from the workbook sheets.
+              Each key represents the sheet title, and its value is a list consisting of:
+              - Tuple containing module and class names extracted from cells (row=2, column=1) and (row=2, column=2).
+              - List of attributes extracted from row 3, starting from column 2.
+              - Dictionary containing row-wise data (starting from row 4, column 2) with column values as per
+               attributes.
+        """
+        extracted_datas = {}
+
+        for sheet in self.workbook.worksheets:
+            class_info = (sheet.cell(row=2, column=1).value, sheet.cell(row=2, column=2).value)
+
+            attributes = []
+            for value in sheet.iter_cols(min_row=3, max_row=3, min_col=2, values_only=True):
+                attributes.append(value[0])
+
+            row_data = {}
+            for i, value in enumerate(sheet.iter_rows(min_row=4, min_col=2, values_only=True)):
+                column_values = []
+                for j, _ in enumerate(value):
+                    cell_value = sheet.cell(row=i + 4, column=j + 2)
+                    if cell_value.hyperlink:
+                        column_values.append(cell_value)
+                    else:
+                        column_values.append(cell_value.value)
+                row_data[i] = column_values
+
+            extracted_datas[sheet.title] = ExcelDataExtract(class_info=class_info, attributes=attributes,
+                                                            datas=row_data.copy())
+
+        return ExcelDatasExtracted(extracted_datas=extracted_datas)
+
+    def read_workbook(self):
+        """
+        Read and process data to instantiate objects from the workbook.
+
+        Reads the processed workbook data, instantiates objects, and organizes them based on the provided classes and
+        modules.
+
+        Note:
+        This method does not support processing variables of complex types, such as `List[Tuple[float, float]]`, due to
+        limitations in interpreting complex nested structures within Excel sheets.
+
+        Returns:
+        object: The instantiated object obtained from the main sheet's data.
+        """
+        instantiated_objects = {}
+        data_structure = self.process_workbook()
+        stack = list(data_structure.extracted_datas.items())
+
+        while stack:
+            key, extracted_data = stack.pop()
+
+            if key == self.main_sheet:
+                instantiated_objects = self.instantiate_main_object(instantiated_objects, extracted_data.class_info,
```

Comment:
As your methods here (`instantiate_main_object`, `process_multiple_hypelinks_row`, `process_single_hyperlink_row`, `process_simple_sheet`) all take the 3 args of extracted_data, you should directly pass extracted_data object to each function.

It could even be ExtractedData methods